pkg/controller/infrastructureconfig/sync_specstatus.go

```diff
@@ -0,0 +1,195 @@
+	"fmt"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-network-operator/pkg/util/ip"
+	"log"
+)
+
+type specStatusSynchronizer interface {
+	SpecStatusSynchronize(*configv1.Infrastructure) (*configv1.Infrastructure, error)
+}
+
+type specStatusSunchronizer struct{}
+
+func (*specStatusSunchronizer) SpecStatusSynchronize(infraConfig *configv1.Infrastructure) (*configv1.Infrastructure, error) {
+	updatedInfraConfig := infraConfig.DeepCopy()
+
+	var (
+		statusApiVips, statusIngressVips           *[]string
+		specApiVips, specIngressVips               *[]configv1.IP
+		statusMachineNetworks, specMachineNetworks *[]configv1.CIDR
+		elb                                        bool
+	)
+
+	if updatedInfraConfig.Status.PlatformStatus == nil {
+		// This should not happen because of guards at openshift/installer but if by any chance we
+		// arrive at this situation, just exit to avoid errors or panics
+		return updatedInfraConfig, nil
+	}
+
+	switch {
+	case updatedInfraConfig.Status.PlatformStatus.Type == configv1.BareMetalPlatformType:
+		if updatedInfraConfig.Spec.PlatformSpec.BareMetal == nil {
+			// This is migration path from pre-4.16 version in which PlatformSpec was not defined.
+			// We first upgrade the struct to a new one (introduced in 4.16) and populate afterward.
+			log.Print("Detected nil platform spec for baremetal, initializing")
+			updatedInfraConfig.Spec.PlatformSpec.BareMetal = &configv1.BareMetalPlatformSpec{}
+		}
+		if updatedInfraConfig.Status.PlatformStatus.BareMetal == nil {
+			// This is a safeguard against strange platform combinations that do not fill the
+			// PlatformStatus (e.g. UPI under some circumstances).
+			log.Print("Detected nil platform status for baremetal, aborting")
+			return updatedInfraConfig, nil
+		}
+		statusApiVips = &updatedInfraConfig.Status.PlatformStatus.BareMetal.APIServerInternalIPs
+		specApiVips = &updatedInfraConfig.Spec.PlatformSpec.BareMetal.APIServerInternalIPs
+
+		statusIngressVips = &updatedInfraConfig.Status.PlatformStatus.BareMetal.IngressIPs
+		specIngressVips = &updatedInfraConfig.Spec.PlatformSpec.BareMetal.IngressIPs
+
+		statusMachineNetworks = &updatedInfraConfig.Status.PlatformStatus.BareMetal.MachineNetworks
+		specMachineNetworks = &updatedInfraConfig.Spec.PlatformSpec.BareMetal.MachineNetworks
+
+		if updatedInfraConfig.Status.PlatformStatus.BareMetal.LoadBalancer != nil &&
+			updatedInfraConfig.Status.PlatformStatus.BareMetal.LoadBalancer.Type == configv1.LoadBalancerTypeUserManaged {
+			elb = true
+		}
+
+	case updatedInfraConfig.Status.PlatformStatus.Type == configv1.VSpherePlatformType:
+		if updatedInfraConfig.Spec.PlatformSpec.VSphere == nil {
+			log.Print("Detected nil platform spec for vsphere, initializing")
+			updatedInfraConfig.Spec.PlatformSpec.VSphere = &configv1.VSpherePlatformSpec{}
+		}
+		if updatedInfraConfig.Status.PlatformStatus.VSphere == nil {
+			log.Print("Detected nil platformstatus for vSphere, aborting")
```

Comment:
`"platform spec"` and `"platformstatus"`.  I think spaces between might be more consistent, unless there is precedent.

Reply:
Will check, I am definitely not the first one to do it so I will just follow the precedent